meteor/client/ui/Status/UserActivity.tsx

```diff
@@ -26,11 +26,62 @@ interface IUserActionsListProps {
 	renderButtons?: (item: UserActionsLogItem) => React.ReactElement
 }
 
-function prettyPrintJsonString(str: string): string {
-	try {
-		return JSON.stringify(JSON.parse(str), undefined, 4)
-	} catch (_e) {
-		return str
+/**
+ * pretty-prints JSON content, and collapses it if it's too long.
+ */
+function CollapseContent(props: { content: any }): JSX.Element {
+	const [expanded, setExpanded] = useState(false)
+
+	const originalString = useMemo(() => {
+		try {
+			let str = JSON.stringify(JSON.parse(props.content), undefined, 2) // Pretty print JSON
+
+			// If the JSON string is a pretty short one, use a condensed JSON instead:
+			if (str.length < 200) {
+				str = JSON.stringify(JSON.parse(props.content)) // Condensed JSON
```

Comment:
you could avoid doing the JSON.parse for a second time here, if you stored the result of the first in a variable